lib/index.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const lockfile = require('@yarnpkg/lockfile')
+const eol = require('eol')
+const nmtree = require('nmtree')
+const { buildYarnTree, buildNpmTree } = require('./tree')
+
+module.exports = {
+  yarnToNpmCore (yarnLock, name, version, packageDir) {
+    const yarnLockNormalized = eol.lf(yarnLock)
+    const yarnObject = lockfile.parse(yarnLockNormalized).object
+    const nodeModulesTree = nmtree(packageDir)
+    const dependencies = buildNpmTree(nodeModulesTree, yarnObject)
+    return JSON.stringify({
+      name,
+      version,
+      lockfileVersion: 1,
+      requires: true,
+      dependencies
+    })
+  },
+  npmToYarnCore (packageLockFileString, packageDir) {
+    const packageLock = JSON.parse(packageLockFileString)
+    const nodeModulesTree = nmtree(packageDir)
+    const yarnTree = buildYarnTree(nodeModulesTree, packageLock)
+    return lockfile.stringify(yarnTree)
+  }
+}
```

Comment:
I like this change, but I don't like the function names. Could we change them to: `convertYarnToNpm` and `convertNpmToYarn`? What do you think?

Reply:
hmm? the name is from ur
```ts
const { buildYarnTree, buildNpmTree } = require('./tree')
```

or u mean this 2

```
yarnToNpmCore
npmToYarnCore
```

Reply:
Yeah, I don't like the -core suffix.